modules/gapi/test/gapi_render_test.cpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+#include "test_precomp.hpp"
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/gapi/render.hpp>
+#include <opencv2/gapi/own/scalar.hpp>
+
+namespace opencv_test
+{
+
+namespace
+{
+    struct RenderTestFixture : public ::testing::Test
+    {
+        int w     = 30;
+        int h     = 40;
+        int thick = 2;
+        int ff    = cv::FONT_HERSHEY_SIMPLEX;
+        int lt    = LINE_8;
+        double fs = 1;
+
+        cv::Mat               ref_mat {320, 480, CV_8UC3, cv::Scalar::all(0)};
+        cv::Mat               out_mat {320, 480, CV_8UC3, cv::Scalar::all(0)};
+        cv::gapi::own::Scalar color   {0, 255, 0};
+        std::string           text    {"some text"};
+
+    };
+} // namespace
+
+TEST_F(RenderTestFixture, PutText)
+{
+    std::vector<cv::gapi::DrawEvent> events;
+
+    for (int i = 0; i < 5; ++i)
+    {
+        int pos_x = 30 + i * 60;
+        int pos_y = 40 + i * 50;
+
+        cv::putText(ref_mat, text, cv::Point(pos_x, pos_y), ff, fs, cv::Scalar{color[0], color[1], color[2]}, thick);
+        events.emplace_back(cv::gapi::TextEvent{text, pos_x, pos_y, ff, fs, color, thick, lt, false});
+    }
+
+    cv::gapi::render(out_mat, events);
+
+    EXPECT_EQ(0, cv::countNonZero(out_mat != ref_mat));
+}
+
+TEST_F(RenderTestFixture, Rectangle)
```

Comment:
I would suggest to add the third test with both render events

Reply:
And one more test with unsupported draw event.

Reply:
Added test

Reply:
BTW can we avoid hardcoded values directly in the test code (like 5, 30, 40, 60) and make them google test parameters?

Reply:
Thanks for comment, it is really needed, but I plan to do it later.